### PR TITLE
session: only update activation environment when running DRM backend or by explicit request

### PR DESCRIPTION
**Pull request by ahesford**
This is an attempt to address https://github.com/labwc/labwc/issues/1530#issuecomment-1975010257 by only updating the dbus/systemd activation environment when the DRM backend is in use or the `LABWC_UPDATE_ACTIVATION_ENV` environment variable explicitly requests an update.

### CODE REVIEW

File: src/config/session.c [hunk 1]
```diff
@@ -74,14 +77,53 @@ read_environment_file(const char *filename)
 }
 
 static void
-update_activation_env(bool initialize)
+backend_check_drm(struct wlr_backend *backend, void *is_drm)
 {
+	if (is_drm && wlr_backend_is_drm(backend)) {
```
Comment (johanmalm) [inline src/config/session.c:82]: Do we need the `is_drm &&` bit here?

**Comment by johanmalm** [comment]
One minor comment. Otherwise LGTM.
Not tested.
We ought to document this, but looking at docs I think the whole systemctl/dbus thing has slipped through the net, so could include here or tackle it as a separate PR.

File: src/config/session.c [hunk 2]
```diff
@@ -74,14 +77,53 @@ read_environment_file(const char *filename)
...
+static bool
+should_update_activation(struct server *server)
+{
+	assert(server);
+
+	static const char *act_env = "LABWC_UPDATE_ACTIVATION_ENV";
+	char *env = getenv(act_env);
+	if (env) {
+		/* Respect any valid preference from the environment */
+		int enabled = parse_bool(env, -1);
```
Comment (Consolatis) [inline src/config/session.c:96]: Should we add "0" and "1" to `src/common/parse-bool.c` ?

**Comment by Consolatis** [comment]
Another minor comment. Otherwise LGTM as well.

File: src/config/session.c (hunk 2 above)
Comment (johanmalm) [reply src/config/session.c:96]: Good idea. Seems right for environment variables in particular.

**Comment by droc12345** [comment]
I don't set my env vars in the env file, I have a shell script that does the initialization before I fire off a compositor.
It doesn't look like this would conflict with the way I do things, but I'll keep an eye on it.

**Comment by ahesford** [comment]
Minor comments addressed in the latest version; I'll put together some documentation on this behavior.

**Comment by ahesford** [comment]
I took a crack at updating the documentation; comments welcome.

**Comment by Consolatis** [comment]
Perfect. Thanks :)

**Pull request merged**